components/org.wso2.openbanking.cds.consent.extensions/src/main/java/org/wso2/openbanking/cds/consent/extensions/admin/impl/CDSConsentAdminHandler.java

```diff
@@ -124,6 +131,16 @@ public void handleRevoke(ConsentAdminData consentAdminData) throws ConsentExcept
                 final String userID = validateAndGetQueryParam(queryParams, USER_ID);
                 DetailedConsentResource detailedConsentResource = this.consentCoreService.getDetailedConsent(consentID);
                 if (detailedConsentResource != null) {
+                    ArrayList<String> userIDs = (ArrayList<String>) consentAdminData.getQueryParams()
+                            .get(CDSConsentExtensionConstants.USER_ID_KEY_NAME);
+                    // userIDs can be null or empty when the request comes from a CustomerCareOfficer
+                    if (userIDs != null && !userIDs.isEmpty()) {
```

Comment:
when can there be a list of userIDs? because usually revoke is done by a single user right? If there is a scenario when a list of userIDs can come shall we add that as a comment as well?

Reply:
It is because `consentAdminData.getQueryParams()` returns a Map, and and `.get(<key>)` returns an array list.  That doesn't mean that we can have multiple userIDs for revoke. Since `ConsentAdminData` is a common class we cannot change that behaviour for revoke only.
For this scenario (revoking a consent from consent manager) we only have 1 user.